### PR TITLE
Sudhanshu Agrawal's Blog Post

### PR DESCRIPTION
**Pull request by SudhanshuAgrawal27**
Simulating the motion of a system of 100 charged bodies by automatic differentiation of the potential energy function.

### CODE REVIEW

**Comment by MikeInnes** [comment]
The post looks good, but there's a bunch of formatting/HTML/`&nbsp`/extra newlines that ideally shouldn't be here. The post should stick to the basic formatting the site provides, and if it doesn't look good we can of course change it, but we should do that for all posts.

**Comment by SudhanshuAgrawal27** [comment]
@MikeInnes What I've done is remove some of the HTML tags like to underline the headings and I've removed some extra &nbsp tags. But I'd still like to center the headings, subheadings and pictures since the post doesn't look as good otherwise. I'd also like to have an &nbsp after each paragraph as I've found that it increases readability. The text seems too dense otherwise.

**Comment by MikeInnes** [comment]
Ok, how about doing it with a small CSS snippet rather than inline then. Then it's easier to fix if we change the global styling later (otherwise it'll end up looking way worse).

**Comment by SudhanshuAgrawal27** [comment]
I'm not quite sure how that's done. Can you give me an example?

**Comment by MikeInnes** [comment]
Should be possible with something like

```html
<style>
h1 {
  text-align: center;
}
</style>
```
at the top of the file.

**Comment by SudhanshuAgrawal27** [comment]
@MikeInnes I think this should work. I've replaced pretty much all the HTML formatting with CSS.

**Comment by MikeInnes** [comment]
you don't need to use HTML tags for the CSS to work -- `#` still means H1, etc. It'd be good to get rid of all the HTML apart from the styling.

**Comment by SudhanshuAgrawal27** [comment]
@MikeInnes I've replaced all the h1 headings with # 
I've styled the h3 and h4 headings which are used twice each. The other places I've used html are to create new lines (<br>) and to create subscript and super script.

**Comment by MikeInnes** [comment]
What I mean is, you can use `#` for H1, `##` for H2, etc, and the styling will still work, since markdown just converts those things into the equivalent HTML tags.

Why are the newlines necessary?

**Comment by SudhanshuAgrawal27** [comment]
@MikeInnes Oh I see. I'll change that.

They're just there to improve readability. I've found that the text seems a bit dense without them. If it truly is going to create a problem I'll remove them.

**Comment by SudhanshuAgrawal27** [comment]
@MikeInnes This will take care of the heading issue. Should I remove all the newlines as well?

**Comment by MikeInnes** [comment]
If there's an issue with text being too dense, I'd prefer to fix that for all posts at once with the site's CSS. So I think just remove them, and I'll tweak the styling after it's merged.

**Comment by SudhanshuAgrawal27** [comment]
@MikeInnes It's done. The places I had issues with is between the end of one paragraph and the heading of another.

**Comment by SudhanshuAgrawal27** [comment]
@MikeInnes The only two places I'd like to have a newline now(assuming you'll tweak the paragraph spacing if you think it's too dense) are between the quote I've used and the author of the quote, and also between my name and email address in the author tag of the blog, since they both look odd without a newline. How can I make that happen?

**Comment by SudhanshuAgrawal27** [comment]
@MikeInnes is this ready to be merged?

File: _posts/2019-09-11-simulating-the-motion-of-charges.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+ 
+
+
+# Conclusion
+
+It works! 
+I created this simulation because it’s an interesting and very visual way to get a real feel of how charges move around when affected by almost a 100 different forces. 
+It also shows how it’s possible to solve a problem more easily by reframing the question being asked. 
+~~Simulate the effect of force on charged bodies?~~
```
Comment (MikeInnes) [inline _posts/2019-09-11-simulating-the-motion-of-charges.md]: Is this meant to be there?

**Comment by SudhanshuAgrawal27** [comment]
@MikeInnes if you're referring to the strike-through then yes, it's meant to be there. It's there to give a visualization to the overall approach of the solution.

**Comment by MikeInnes** [comment]
The headings looks totally fine if you use the right ones. The bigger issues for readability were things like inconsistent formatting, having to scroll horizontally for each code example etc (could the comments be part of the main text instead?). But I've tried to fix what I can of that, so I think it looks OK now and only uses markdown.

**Comment by SudhanshuAgrawal27** [comment]
@MikeInnes 
I don't have a way of running the site with the new changes. But I've looked them over and everything seems fine.

File: _posts/2019-09-11-simulating-the-motion-of-charges.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+It’s safe to say that the system has converged at a value of approximately -1360 .
+
+You’re probably wondering why the graph is so bumpy and what these extreme minimas are. The minimas are simply the points in the training loop when the charges got a bit overconfident and tried to escape the constraining circle. They are then immediately brought back to valid positions by the loss function.
+They may also occur when a positive and a negative charge briefly overlap with each other giving a potential of negative infinity. Of course, such a situation is impossible in the real world.
+
+## Conclusion
+
+It works!
+I created this simulation because it’s an interesting and very visual way to get a real feel of how charges move around when affected by almost a 100 different forces.
```
Comment (MikeInnes) [inline _posts/2019-09-11-simulating-the-motion-of-charges.md]: This should be either "a hundred" or "100", but not "a 100".

**Pull request merged**